scripts/retry.js

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env node
+
+import * as childProcess from 'child_process';
+
+/*
+ * This retry script is intentionally written in pure JavaScript and
+ * uses no dependencies other than node built-ins.
+ * This is to make sure that script can be run without requiring
+ * 'npm install'.
+ *
+ * Usage:
+ * ./scripts/retry.js <command>
+ * ./scripts.retry.js --maxAttempts <number> <command>
+ */
+
+const usage =
+  'Usage:\n./scripts/retry.js <command>\n./scripts.retry.js --maxAttempts <number> <command>';
+
+class RetryOptions {
+  constructor(maxAttempts, command) {
+    this.maxAttempts = maxAttempts;
+    this.command = command;
+  }
+}
+
+const parseInput = () => {
```

Comment:
We can use the built-in [parseArgs](https://nodejs.org/api/util.html#utilparseargsconfig) utility to make args parsing a little cleaner / more robust

Reply:
I'll try. If this can surface the vararg-ish tail (the command part) nicely then I'll use it.

Reply:
https://github.com/aws-amplify/amplify-backend/pull/1233/commits/02a8f092dd1500d0ad01e28d2e35fa6e8a75048e